File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
  for (let i=1; i<=Math.floor(Math.sqrt(yellow)); i++) {
    if (yellow%i!=0) continue;
    let j = yellow/i;
    if (brown == 4+(i+j)*2) {
      return [j+2, i+2];
    }
  }
}

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int width = 0;
        int height = 0;
        int[] answer = new int[2];
        
        for (int i=1; i<=Math.sqrt(yellow); i++) {
            if (yellow % i == 0) {
                width = yellow / i;
                height = i;
                if (brown == (width + height) * 2 + 4) {
                    answer[0] = width+2;
                    answer[1] = height+2;
                    break;
                }
            }
        }
        
        
        return answer;
    }
}